src/Symfony/Component/Validator/Constraints/Email.php

```
@@ -27,6 +27,7 @@ class Email extends Constraint
 {
     public const VALIDATION_MODE_HTML5 = 'html5';
     public const VALIDATION_MODE_STRICT = 'strict';
+    /** @deprecated since Symfony 5.4; will be removed in Symfony 6.0; use email validation mode "html5" instead */
     public const VALIDATION_MODE_LOOSE = 'loose';
```

To actually be deprecated, the constraint should also trigger a deprecation when you instantiate it with a config putting it in loose mode. It should not be deprecated only for the _default_ mode

---

Yeah sure, but I thought this line would take care of this!?:
https://github.com/symfony/symfony/blob/67a2150c11f02b2d974cb48ca04577c218063cdf/src/Symfony/Component/Validator/Constraints/EmailValidator.php#L43

---

this line is only about the *default* mode of the validator, not about the mode configured in the constraint (which wins over the default mode)

---

But isn't the actual mode being passed to `__construct()`? I thought `$defaultMode` was just a somewhat misleading name for the argument...

---

`$defaultMode` is in the constructor of the validator, not of the constraint. It is the mode used by default when the constraint does not configure its mode (see the `null === $constraint->mode` check in the validator)

---

OK, but isn't the line we just moved upwards doing what you're asking for?:
https://github.com/symfony/symfony/blob/fd6741c4288f0d29a782030c3baa2959a65dfbdc/src/Symfony/Component/Validator/Constraints/EmailValidator.php#L76

Or do you suggest a similar check here?:
https://github.com/symfony/symfony/blob/fd6741c4288f0d29a782030c3baa2959a65dfbdc/src/Symfony/Component/Validator/Constraints/Email.php#L71

... but wouldn't this then raise *two* deprecation warnings, as @nicolas-grekas said below?

Sorry guys, I'm a little confused... ;-)